app/src/lib/stores/app-store.ts

```diff
@@ -5016,40 +5018,30 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
-  /** This shouldn't be called directly. See `Dispatcher`. */
-  public async _createStash(
-    repository: Repository,
-    branchName: string,
-    removePreviousStash: boolean = true
-  ) {
+  /**
+   *  Note: **drops the existing desktop stash** on that branch (if there is one)
+   */
+  private async createStash(repository: Repository, branchName: string) {
     if (!enableStashing()) {
       return
     }
-    const { branchesState } = this.repositoryStateCache.get(repository)
-    const { tip } = branchesState
-    if (tip.kind !== TipState.Valid) {
-      return
-    }
 
     // get the previous stash before we create a new one
-    const previousStash = removePreviousStash
-      ? await getLastDesktopStashEntryForBranch(repository, branchName)
-      : null
+    const previousStash = await getLastDesktopStashEntryForBranch(
```

Comment:
Removing the parameter, `removePreviousStash` is a bit important here. We only want to get the last stash to drop if we are stashing changes on the current branch because we want one stash per branch. Now that this parameter is removed, the previous stash could be deleted when bringing changes to the new branch. This is bad because we could be deleting a stash that was explicitly stashed on another branch when bringing changes to it from some other branch. Another downside to removing this parameter is now this relatively expensive call will execute every-time, even in cases where it shouldn't as described above.

Reply:
thanks @niik! yep that's exactly right :)

Reply:
I did something similar to that in eba6310081a54e784af3f328e41f60d402a8e77b. Moving that logic into a callback the caller supplies.